### ISSUE TITLE
LR schedules

### ISSUE THREAD

**Issue by alxndrTL** [Haiyang-W/TokenFormer#11]
Hello,
Thank you for open-sourcing the training code for TokenFormer.

As I understand, you have yet to release the incremental training code, but I was wondering, do you treat each training independently like a new training (the only exception being taking the weights of the prev model) ?
For example, when is the LR decayed? Once per model size? Or once for the whole training ?

Thank you

**Comment by Haiyang-W** [comment]
Sorry for the late reply. You’ll need to experiment with the learning rate (LR) yourself. That means I treat each training independently. From my tests, using the same LR tends to slightly work better, although the loss will increase at the beginning. If you're using the same dataset for training and originally used a cosine LR schedule, I also suggest adding some noise to the old parameters. This can also help stabilize the training process. 

You can also experiment with different initializations for K and V. Whether you initialize them partially with zeros or entirely with random values, the final performance is roughly the same. For fixed datasets (without new data) and cosine LR schedules, random initialization tends to result in more stable training, but the final loss is quite similar.
